File: Pertemuan-13/Tugas/AppPerpus/Buku.py
import requests
import json
class Buku:
    def __init__(self):
        self.__id=None
        self.__kodebuku = None
        self.__judul = None
        self.__penulis = None
        self.__penerbit = None
        self.__tahunpenerbit = None
        self.__stok = None
        self.__url = "http://f0832375.xsph.ru/app/buku_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kodebuku(self):
        return self.__kodebuku
        
    @kodebuku.setter
    def kodebuku(self, value):
        self.__kodebuku = value
    @property
    def judul(self):
        return self.__judul
        
    @judul.setter
    def judul(self, value):
        self.__judul = value
    @property
    def penulis(self):
        return self.__penulis
        
    @penulis.setter
    def penulis(self, value):
        self.__penulis = value
    @property
    def penerbit(self):
        return self.__penerbit
        
    @penerbit.setter
    def penerbit(self, value):
        self.__penerbit = value
    @property
    def tahunpenerbit(self):
        return self.__tahunpenerbit
        
    @tahunpenerbit.setter
    def tahunpenerbit(self, value):
        self.__tahunpenerbit = value
    @property
    def stok(self):
        return self.__stok
        
    @stok.setter
    def stok(self, value):
        self.__stok = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kodebuku(self, kodebuku):
        url = self.__url+"?kodebuku="+kodebuku
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id']
            self.__kodebuku = item['kodebuku']
            self.__judul = item['judul']
            self.__penulis = item['penulis']
            self.__penerbit = item['penerbit']
            self.__tahunpenerbit = item['tahunpenerbit']
            self.__stok = item['stok']
        return data
    def simpan(self):
        payload = {
            "kodebuku":self.__kodebuku,
            "judul":self.__judul,
            "penulis":self.__penulis,
            "penerbit":self.__penerbit,
            "tahunpenerbit":self.__tahunpenerbit,
            "stok":self.__stok
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kodebuku(self, kodebuku):
        url = self.__url+"?kodebuku="+kodebuku
        payload = {
            "kodebuku":self.__kodebuku,
            "judul":self.__judul,
            "penulis":self.__penulis,
            "penerbit":self.__penerbit,
            "tahunpenerbit":self.__tahunpenerbit,
            "stok":self.__stok
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kodebuku(self,kodebuku):
        url = self.__url+"?kodebuku="+kodebuku
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: Pertemuan-12/AppPerpus/Buku.php
<?php
//Simpanlah dengan nama file : Buku.php
require_once 'database.php';
class Buku 
{
    private $db;
    private $table = 'buku';
    public $kodebuku = "";
    public $judul = "";
    public $penulis = "";
    public $penerbit = "";
    public $tahunpenerbit = "";
    public $stok = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kodebuku(int $kodebuku)
    {
        $query = "SELECT * FROM $this->table WHERE kodebuku = $kodebuku";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kodebuku`,`judul`,`penulis`,`penerbit`,`tahunpenerbit`,`stok`) VALUES ('$this->kodebuku','$this->judul','$this->penulis','$this->penerbit','$this->tahunpenerbit','$this->stok')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kodebuku = '$this->kodebuku', judul = '$this->judul', penulis = '$this->penulis', penerbit = '$this->penerbit', tahunpenerbit = '$this->tahunpenerbit', stok = '$this->stok' 
        WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kodebuku($kodebuku): int
    {
        $query = "UPDATE $this->table SET kodebuku = '$this->kodebuku', judul = '$this->judul', penulis = '$this->penulis', penerbit = '$this->penerbit', tahunpenerbit = '$this->tahunpenerbit', stok = '$this->stok' 
        WHERE kodebuku = $kodebuku";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kodebuku($kodebuku): int
    {
        $query = "DELETE FROM $this->table WHERE kodebuku = $kodebuku";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>